CI fails with dev dependencies: RuntimeError: expand(CUDABFloat16Type{[3, 1, 1, 8]}, size=[-1, -1, 4]): the number of sizes provided (3) must be greater or equal to the number of dimensions in the tensor (4)

CI fails with dev dependencies: https://github.com/huggingface/trl/actions/runs/30137414569/job/89624014394
> RuntimeError: expand(CUDABFloat16Type{[3, 1, 1, 8]}, size=[-1, -1, 4]): the number of sizes provided (3) must be greater or equal to the number of dimensions in the tensor (4)

```python
FAILED tests/test_grpo_trainer.py::TestGRPOTrainer::test_train[trl-internal-testing/tiny-NemotronHForCausalLM-nano] - RuntimeError: expand(CUDABFloat16Type{[3, 1, 1, 8]}, size=[-1, -1, 4]): the number of sizes provided (3) must be greater or equal to the number of dimensions in the tensor (4)
FAILED tests/test_rloo_trainer.py::TestRLOOTrainer::test_train[trl-internal-testing/tiny-NemotronHForCausalLM-nano] - RuntimeError: expand(CUDABFloat16Type{[3, 1, 1, 8]}, size=[-1, -1, 4]): the number of sizes provided (3) must be greater or equal to the number of dimensions in the tensor (4)
  = 2 failed
```

Traceback:
```python
  >       trainer.train()
  
  tests/test_grpo_trainer.py:330: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1448: in train
      return inner_training_loop(
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1530: in _inner_training_loop
      self._run_epoch(
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1758: in _run_epoch
      tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1549: in training_step
      output = super().training_step(model, inputs, num_items_in_batch)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/trainer.py:1924: in training_step
      inputs = self._prepare_inputs(inputs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/extras/profiling.py:211: in wrapper
      return func(self, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1578: in _prepare_inputs
      generation_batch = self._generate_and_score_completions(generation_batch)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:2387: in _generate_and_score_completions
      ) = self._generate(prompts)
          ^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:2188: in _generate
      completion_ids, logprobs = self._generate_single_turn(prompt_ids, images, multimodal_fields)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  trl/trainer/grpo_trainer.py:1877: in _generate_single_turn
      prompt_completion_ids = unwrapped_model.generate(
  .venv/lib/python3.12/site-packages/torch/utils/_contextlib.py:124: in decorate_context
      return func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/generation/utils.py:2651: in generate
      result = decoding_method(
  .venv/lib/python3.12/site-packages/transformers/generation/utils.py:2865: in _sample
      outputs = model_forward(**model_inputs, return_dict=True)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1778: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1789: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/accelerate/utils/operations.py:943: in forward
      return model_forward(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/accelerate/utils/operations.py:931: in __call__
      return convert_to_fp32(self.model_forward(*args, **kwargs))
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/amp/autocast_mode.py:44: in decorate_autocast
      return func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/utils/generic.py:939: in wrapper
      output = func(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/nemotron_h/modeling_nemotron_h.py:1193: in forward
      outputs = self.model(
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1778: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1789: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/utils/generic.py:1068: in wrapper
      output = func(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/utils/output_capturing.py:262: in wrapper
      outputs = func(self, *args, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/nemotron_h/modeling_nemotron_h.py:1127: in forward
      hidden_states = mixer_block(
  .venv/lib/python3.12/site-packages/transformers/modeling_layers.py:114: in __call__
      return super().__call__(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1778: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1789: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/nemotron_h/modeling_nemotron_h.py:963: in forward
      hidden_states = self.mixer(hidden_states, cache_params=past_key_values, attention_mask=attention_mask)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1778: in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/torch/nn/modules/module.py:1789: in _call_impl
      return forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/integrations/accelerate.py:941: in wrapped
      output = forward_func(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  .venv/lib/python3.12/site-packages/transformers/models/nemotron_h/modeling_nemotron_h.py:605: in forward
      return self.cuda_kernels_forward(hidden_states, cache_params, attention_mask)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  self = NemotronHMamba2Mixer(
    (act): SiLUActivation()
    (conv1d): Conv1d(64, 64, kernel_size=(4,), stride=(1,), padding=(3,)...res=104, bias=False)
    (norm): Zamba2RMSNormGated()
    (out_proj): Linear(in_features=32, out_features=16, bias=False)
  )
  hidden_states = tensor([[[-5.4932e-02, -1.2451e-02,  5.1880e-03,  1.1780e-02, -6.8665e-03,
            -2.3315e-02, -3.7842e-02, -3.9062...2,  4.1504e-02, -3.7842e-02,  1.4587e-02,
            -4.9561e-02, -1.3855e-02]]], device='cuda:0', dtype=torch.bfloat16)
  cache_params = DynamicCache(layers=[LinearAttentionLayer, DynamicLayer, LinearAttentionLayer])
  attention_mask = None, kwargs = {}
  
      def cuda_kernels_forward(
          self,
          hidden_states: torch.Tensor,
          cache_params: Cache | None = None,
          attention_mask: torch.Tensor | None = None,
          **kwargs,
      ):
          # 1. Gated MLP's linear projection
          hidden_states = apply_mask_to_padding_states(hidden_states, attention_mask)
          projected_states = self.in_proj(hidden_states)
      
          A = -torch.exp(self.A_log.float())
          dt_limit_kwargs = {} if self.time_step_limit == (0.0, float("inf")) else {"dt_limit": self.time_step_limit}
          # Fused kernel for conv1d, SSM, and the final projection
          if self.training and cache_params is None:
              return mamba_split_conv1d_scan_combined(
                  projected_states,
                  self.conv1d.weight.squeeze(1),
                  self.conv1d.bias,
                  self.dt_bias,
                  A,
                  D=self.D,
                  chunk_size=self.chunk_size,
                  seq_idx=kwargs.get("seq_idx"),
                  activation=self.activation,
                  rmsnorm_weight=self.norm.weight,
                  rmsnorm_eps=self.norm.variance_epsilon,
                  outproj_weight=self.out_proj.weight,
                  outproj_bias=self.out_proj.bias,
                  headdim=self.head_dim,
                  ngroups=self.n_groups,
                  norm_before_gate=False,
                  return_final_states=False,
                  **dt_limit_kwargs,
              )
      
          # Set up dimensions for reshapes later
          batch_size, seq_len, _ = hidden_states.shape
          groups_time_state_size = self.n_groups * self.ssm_state_size
          use_precomputed_states = cache_params is not None and cache_params.has_previous_state(self.layer_idx)
      
          gate, hidden_states_B_C, dt = projected_states.split(
              [self.intermediate_size, self.conv_dim, self.num_heads], dim=-1
          )
      
          # Apply the conv
          hidden_states_B_C = self._convolution(hidden_states_B_C, cache_params, attention_mask, **kwargs)
          hidden_states_B_C = apply_mask_to_padding_states(hidden_states_B_C, attention_mask)
          hidden_states, B, C = torch.split(
              hidden_states_B_C,
              [self.intermediate_size, groups_time_state_size, groups_time_state_size],
              dim=-1,
          )
      
          recurrent_state = cache_params.layers[self.layer_idx].recurrent_states[0] if use_precomputed_states else None
          # Single step calculations via cache
          if use_precomputed_states and seq_len == 1:
              # 3. SSM transformation
              A = A[:, None, ...][:, :, None].expand(-1, self.head_dim, self.ssm_state_size).to(dtype=torch.float32)
  >           dt = dt[:, :, None].expand(-1, -1, self.head_dim)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  E           RuntimeError: expand(CUDABFloat16Type{[3, 1, 1, 8]}, size=[-1, -1, 4]): the number of sizes provided (3) must be greater or equal to the number of dimensions in the tensor (4)
  
  .venv/lib/python3.12/site-packages/transformers/models/nemotron_h/modeling_nemotron_h.py:371: RuntimeError
```

--- a/tests/test_grpo_trainer.py
+++ b/tests/test_grpo_trainer.py
@@ -35,7 +35,7 @@ from transformers import (
     BitsAndBytesConfig,
 )
 from transformers.testing_utils import backend_empty_cache, torch_device
-from transformers.utils import is_kernels_available, is_peft_available
+from transformers.utils import is_peft_available
 
 from trl import GRPOConfig, GRPOTrainer
 from trl.import_utils import is_liger_kernel_available
@@ -300,20 +300,10 @@ class TestGRPOTrainer(TrlTestCase):
             "trl-internal-testing/tiny-Qwen3MoeForCausalLM",
             pytest.param(
                 "trl-internal-testing/tiny-NemotronHForCausalLM-nano",
-                marks=[
-                    pytest.mark.skipif(
-                        Version(transformers.__version__) < Version("5.7.0"),
-                        reason="Nemotron 3 gradient checkpointing requires transformers>=5.7.0 (see transformers#45625)",
-                    ),
-                    pytest.mark.xfail(
-                        Version(transformers.__version__).is_devrelease and is_kernels_available(),
-                        reason=(
-                            "transformers' NemotronHMamba2Mixer.cuda_kernels_forward breaks generation when Hub kernels "
-                            "are available (see #6541)."
-                        ),
-                        strict=True,
-                    ),
-                ],
+                marks=pytest.mark.skipif(
+                    Version(transformers.__version__) < Version("5.7.0"),
+                    reason="Nemotron 3 gradient checkpointing requires transformers>=5.7.0 (see transformers#45625)",
+                ),
             ),
         ],
     )

--- a/tests/test_rloo_trainer.py
+++ b/tests/test_rloo_trainer.py
@@ -25,7 +25,7 @@ from transformers import (
     AutoModelForSequenceClassification,
     AutoTokenizer,
 )
-from transformers.utils import is_kernels_available, is_peft_available
+from transformers.utils import is_peft_available
 
 from trl import RLOOConfig, RLOOTrainer
 
@@ -62,20 +62,10 @@ class TestRLOOTrainer(TrlTestCase):
             "trl-internal-testing/tiny-Qwen3MoeForCausalLM",
             pytest.param(
                 "trl-internal-testing/tiny-NemotronHForCausalLM-nano",
-                marks=[
-                    pytest.mark.skipif(
-                        Version(transformers.__version__) < Version("5.7.0"),
-                        reason="Nemotron 3 gradient checkpointing requires transformers>=5.7.0 (see transformers#45625)",
-                    ),
-                    pytest.mark.xfail(
-                        Version(transformers.__version__).is_devrelease and is_kernels_available(),
-                        reason=(
-                            "transformers' NemotronHMamba2Mixer.cuda_kernels_forward breaks generation when Hub kernels "
-                            "are available (see #6541)."
-                        ),
-                        strict=True,
-                    ),
-                ],
+                marks=pytest.mark.skipif(
+                    Version(transformers.__version__) < Version("5.7.0"),
+                    reason="Nemotron 3 gradient checkpointing requires transformers>=5.7.0 (see transformers#45625)",
+                ),
             ),
         ],
     )

